apache/superset
# feat: Implement ability to disable control sections

### SUMMARY
Currently it is only possible to hide specific controls in Explore but not to disable entire control sections. This PR introduces the ability to disable entire control sections in Explore based on given rules.

With the newly implemented ability, this PR disables/enables the Predictive Analytics section in Explore based on whether the x axis column is temporal or not.

Closes https://github.com/apache/superset/pull/21975

### AFTER

https://user-images.githubusercontent.com/60598000/207888212-dcc64368-2651-47a3-a2e9-d02e39a21278.mp4

### TESTING INSTRUCTIONS
1. Enable GENERIC_CHART_AXES
2. Open a Time-series chart for example
3. Observe the Predictive Analytics sections being disabled and enabled as you change the X-Axis column

### ADDITIONAL INFORMATION
